processing/src/main/java/io/druid/query/extraction/ExtractionFn.java

```diff
@@ -29,7 +30,9 @@
     @JsonSubTypes.Type(name = "partial", value = MatchingDimExtractionFn.class),
     @JsonSubTypes.Type(name = "searchQuery", value = SearchQuerySpecDimExtractionFn.class),
     @JsonSubTypes.Type(name = "javascript", value = JavascriptExtractionFn.class),
-    @JsonSubTypes.Type(name = "timeFormat", value = TimeFormatExtractionFn.class)
+    @JsonSubTypes.Type(name = "timeFormat", value = TimeFormatExtractionFn.class),
+    @JsonSubTypes.Type(name = "namespace", value = NamespacedExtraction.class),
+    @JsonSubTypes.Type(name = "explicitRename", value = ExplicitDimRenameFn.class)
```

Comment:
why not make lookups share the same extraction function, taking either a map or a namespace as configuration, removing the need for an additional extraction function?